packages/scm/src/browser/dirty-diff/dirty-diff-widget.ts

```diff
@@ -0,0 +1,387 @@
+// *****************************************************************************
+
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { ActionMenuNode, Disposable, Event, MenuCommandExecutor, MenuModelRegistry, MenuPath, URI, nls } from '@theia/core';
+import { ContextKeyService } from '@theia/core/lib/browser/context-key-service';
+import { MonacoEditor } from '@theia/monaco/lib/browser/monaco-editor';
+import { ChangeRangeMapping, LineRange, NormalizedEmptyLineRange } from './diff-computer';
+import { ScmColors } from '../scm-colors';
+import * as monaco from '@theia/monaco-editor-core';
+import { PeekViewWidget, peekViewBorder, peekViewTitleBackground, peekViewTitleForeground, peekViewTitleInfoForeground }
+    from '@theia/monaco-editor-core/esm/vs/editor/contrib/peekView/browser/peekView';
+import { StandaloneServices } from '@theia/monaco-editor-core/esm/vs/editor/standalone/browser/standaloneServices';
+import { IInstantiationService } from '@theia/monaco-editor-core/esm/vs/platform/instantiation/common/instantiation';
+import { ICodeEditor } from '@theia/monaco-editor-core/esm/vs/editor/browser/editorBrowser';
+import { IPosition, Position } from '@theia/monaco-editor-core/esm/vs/editor/common/core/position';
+import { IRange, Range } from '@theia/monaco-editor-core/esm/vs/editor/common/core/range';
+import { IDiffEditorOptions } from '@theia/monaco-editor-core/esm/vs/editor/common/config/editorOptions';
+import { EmbeddedDiffEditorWidget } from '@theia/monaco-editor-core/esm/vs/editor/browser/widget/embeddedCodeEditorWidget';
+import { ITextModelService } from '@theia/monaco-editor-core/esm/vs/editor/common/services/resolverService';
+import { Action, IAction } from '@theia/monaco-editor-core/esm/vs/base/common/actions';
+import { Codicon } from '@theia/monaco-editor-core/esm/vs/base/common/codicons';
+import { ThemeIcon } from '@theia/monaco-editor-core/esm/vs/base/common/themables';
+import { ScrollType } from '@theia/monaco-editor-core/esm/vs/editor/common/editorCommon';
+import { Color } from '@theia/monaco-editor-core/esm/vs/base/common/color';
+import { IColorTheme, IThemeService } from '@theia/monaco-editor-core/esm/vs/platform/theme/common/themeService';
+
+export const SCM_CHANGE_TITLE_MENU: MenuPath = ['scm-change-title-menu'];
+/** Reserved for plugin contributions, corresponds to contribution point 'scm/change/title'. */
+export const PLUGIN_SCM_CHANGE_TITLE_MENU: MenuPath = ['plugin-scm-change-title-menu'];
+
+export const DirtyDiffWidgetProps = Symbol('DirtyDiffWidgetProps');
+export interface DirtyDiffWidgetProps {
+    readonly editor: MonacoEditor;
+    readonly previousRevisionUri: URI;
+    readonly changes: readonly ChangeRangeMapping[];
+}
+
+export const DirtyDiffWidgetFactory = Symbol('DirtyDiffWidgetFactory');
+export type DirtyDiffWidgetFactory = (props: DirtyDiffWidgetProps) => DirtyDiffWidget;
+
+@injectable()
+export class DirtyDiffWidget implements Disposable {
```

Comment:
It seems I missed the connotation that "widget" has in Theia... I'm open to suggestions for a better name.

My initial proposal would be to rename the internal class `DirtyDiffPeekView` to `DirtyDiffPeekViewWidget` (since it extends `PeekViewWidget`), and then rename the exported class `DirtyDiffWidget` to `DirtyDiffPeekView`. Unfortunately, it would also affect the names of the related types and variables...